sources/folderUtils.ts

```diff
@@ -52,6 +53,8 @@ export function getTemporaryFolder(target: string = tmpdir()) {
     } catch (error) {
       if ((error as NodeError).code === `EEXIST`) {
         continue;
+      } else if ((error as NodeError).code === `EACCES`) {
+        throw new UsageError(`Failed to create cache directory. Please ensure the user has write access to the target directory (${target}). If the user's home directory does not exist, create it first.`, {cause: error});
```

Comment:
@arcanis it would maybe make sense to add support for this in Clipanion, no?

Reply:
```suggestion
        // @ts-expect-error UsageError expects one argument
        throw new UsageError(`Failed to create cache directory. Please ensure the user has write access to the target directory (${target}). If the user's home directory does not exist, create it first.`, {cause: error});
```

Reply:
For now should we remove the `cause: error`?